### PR TITLE
starting refactor

### PR DESCRIPTION
**Pull request by Adrianbakke**

### CODE REVIEW

File: tinygrad/ops/ops_gpu.py [hunk 1]
```diff
@@ -2,9 +2,16 @@
 import numpy as np
 from tinygrad.helpers import binary_broadcast
 from ..tensor import Function
-from ..llops.gpu import GPUBuffer
-from ..llops.gpu import unary_op, binary_op, reduce_op, perm_axis, inner_slice
-from ..llops.gpu import matmul, conv, convdw, convdx
+
+import os
```
Comment (Adrianbakke) [inline tinygrad/ops/ops_gpu.py:6]: Is it something in the gist of this you have in mind? The goal is to only use ops_gpu and get rid ops_cuda file? @geohot

File: tinygrad/ops/ops_gpu.py [hunk 2]
```diff
@@ -2,9 +2,16 @@
...
+if os.getenv("CUDA")==1:
+  from ..llops.cuda import GPUBuffer
+  from ..llops.cuda import unary_op, binary_op, reduce_op, perm_axis, inner_slice
+  from ..llops.cuda import matmul, conv, convdw, convdx
+else:
+  from ..llops.gpu import GPUBuffer
+  from ..llops.gpu import unary_op, binary_op, reduce_op, perm_axis, inner_slice
+  from ..llops.gpu import matmul, conv, convdw, convdx
```
Comment (Adrianbakke) [inline tinygrad/ops/ops_gpu.py:14]: Is it something in the gist of this you have in mind? The goal is to only use ops_gpu and get rid ops_cuda file? @geohot

**Comment by geohot** [comment]
This still alive?

**Comment by Adrianbakke** [comment]
There were alot of changes to the codebase at the time, so I figured I wait until things started to settle. I will pick it back up

**Comment by geohot** [comment]
Closing for now. It's probably best to wait on the CUDA backend for LLVM to be finished, there's a lot to reuse there.

**Pull request closed without merging**